addons/l10n_id_efaktur_coretax/models/account_move.py

```diff
@@ -0,0 +1,307 @@
+
+        Change the flow of efaktur downloading. Collects data needed for efaktur and generate the
+        xml file.
+        """
+        # Pre-download checks
+        err_message = _("Unable to download E-Faktur for the following reason(s):\n")
+        for record in self:
+            if record.state == 'draft':
+                err_message += _('- Invoice is in draft state\n')
+            if not record.country_code == 'ID':
+                err_message += _("- Invoice is not under Indonesian company \n")
+            if not record.partner_id.commercial_partner_id.l10n_id_pkp:
+                err_message += _("- Customer %s is not taxable, tick ID PKP if necessary\n", record.partner_id.commercial_partner_id.name)
+            if not record.move_type == 'out_invoice':
+                err_message += _("- This is not an invoice\n")
+            if not record.line_ids.tax_ids:
+                err_message += _("- Invoice does not contain any taxes\n")
+            if record.partner_id.l10n_id_buyer_document_type != 'TIN' and not record.partner_id.l10n_id_buyer_document_number:
+                err_message += _("- Document number for customer hasn't been filled in\n")
+            if record.l10n_id_kode_transaksi == "07":
+                if not (record.l10n_id_coretax_add_info_07 and record.l10n_id_coretax_facility_info_07):
+                    err_message += _("- Invoice doesn't contain the Additional info and Facility Stamp yet, (Kode 07)\n")
+            if record.l10n_id_kode_transaksi == "08":
+                if not (record.l10n_id_coretax_add_info_08 and record.l10n_id_coretax_facility_info_08):
+                    err_message += _("- Invoice doesn't contain the Additional info and Facility Stamp yet, (Kode 08)\n")
+            if not record.partner_id.vat:
+                err_message += _("- NPWP for customer %s hasn't been filled in yet\n", record.partner_id.name)
+            if not record.company_id.vat:
+                err_message += _("- Your comapny's VAT hasn't been configured yet\n")
+
+        if err_message:
+            raise ValidationError(err_message)
+        # Should prevent users from generating e-Faktur document on invoices across multi-company.
+        # Allowing it will cause issues on the invoice/eFaktur document record rule
+        if len(self.company_id) > 1:
+            raise UserError(_("You are not allowed to generate e-Faktur document from invoices coming from different companies"))
+
+        # All invoices in self have no documents; we can create a new one for them.
+        # Or all invoices in self have a document, but it's the same one. Special use case but we allow downloading it.
+        if not self.l10n_id_coretax_document:
+            self.l10n_id_coretax_document = self.env['l10n_id_efaktur_coretax.document'].create({
+                'invoice_ids': self.ids,
+                'company_id': self.company_id.id,
+            })
+            self.l10n_id_coretax_document.action_regenerate()
```

Comment:
```suggestion
            self.l10n_id_coretax_document._generate_xml()
```
Will do the same and avoid possible weird bugs if we change regenerate and suddenly see that behavior applied to the first download